test/integration/provider/admin/backend_apis_controller_test.rb

```diff
@@ -53,4 +53,21 @@ class Provider::Admin::BackendApisControllerTest < ActionDispatch::IntegrationTe
     assert_response :redirect
     assert_equal 'https://new-endpoint.com:443/p', backend_api.reload.private_endpoint
   end
+
+  test 'delete a backend api with products' do
+    backend_api = @provider.backend_apis[0]
+    assert backend_api.backend_api_configs.any?
+
+    delete provider_admin_backend_api_path(backend_api)
+    assert_nothing_raised(ActiveRecord::RecordNotFound) { backend_api.reload }
```

Comment:
`assert_nothing_raised` is noop so no need to do it. Not much value and removed in higher version of testunit ;)
You can also assert the redirection

Reply:
_backend_api.reload_ doesn't mean anything from testing perspective. The gist of this test is to not to throw an error when calling `reload`, isn't it? So why not be as explicit as possible for the sake of future readers?

Reply:
Ideally there should be a `assert record.exist` or something like that, but the exception is the only way I could think of

Reply:
you can do `assert backend_api.reload` or  `assert BackendApi.exists?(backend_api.id)`, or:
```ruby
assert_no_difference(BackendApi.method(:count)) do
  delete provider_admin_backend_api_path(backend_api)
end
```

Reply:
Btw, the tests should test the flash message too 😄

Reply:
How can I do that? @Martouta

Reply:
`assert_equal 'the expected message here', flash[:notice]` and `assert_equal 'the expected message here', flash[:error]`

Reply:
Example:
https://github.com/3scale/porta/blob/d57cbd4db56d1c168dfc4be8386f7704f63c9847/test/integration/developer_portal/invitation_signup_test.rb#L69